src/main/java/com/google/cloud/spanner/pgadapter/parsers/DateParser.java

```diff
@@ -114,13 +116,28 @@ protected byte[] binaryParse() {
     if (this.item == null) {
       return null;
     }
-    LocalDate localDate =
-        LocalDate.of(this.item.getYear(), this.item.getMonth(), this.item.getDayOfMonth());
+    return convertToPG(this.item);
+  }
+
+  static byte[] convertToPG(Date value) {
+    LocalDate localDate = LocalDate.of(value.getYear(), value.getMonth(), value.getDayOfMonth());
     long days = localDate.toEpochDay() - PG_EPOCH_DAYS;
     int daysAsInt = validateRange(days);
     return IntegerParser.binaryParse(daysAsInt);
   }
 
+  public static byte[] convertToPG(ResultSet resultSet, int position, DataFormat format) {
+    switch (format) {
+      case SPANNER:
```

Comment:
If the conversion to format is not possible for these data types, can we add some more specific error message rather than falling back to `unknown data format`?

Reply:
(See above)
The default case should not be reachable, but is required for the Java compiler, as it otherwise thinks that there are code paths possible that will not return a value for this method.